### DISCUSSION TITLE
`ruff-report` – a browser tool for slicing and dicing a Ruff report

### DISCUSSION THREAD

**Discussion by akx** [astral-sh/ruff#5078] [Show and tell]
Hey,

I wrote a small React tool for slicing and dicing Ruff JSON reports called, aptly enough, `ruff-report` (and this is what I needed https://github.com/astral-sh/ruff/pull/5059 for 😁).

This was really a "scratch-my-own-itch" thing – when you slap Ruff on an existing codebase with `--select=ALL`, it's nice to be able to see a summary of issues that might be easy pickings for `select`ing into the configuration, and then go from there. 

You can filter the full report by package, module, code class, fixability, etc. for a better view, as it were.

I figured it might be useful for other folks too! It's probably a bit <del>ruff</del> rough around the edges (especially don't try with any old random JSON file, it'll probably crash on you), and the design is really just [Mantine](https://mantine.dev/)'s defaults. PRs and ideas are very welcome!

## Links

* [Try it out!](https://akx.github.io/ruff-report/)
* [GitHub repo](https://github.com/akx/ruff-report)

## Screenshots

![Screenshot 2023-06-14 at 10 54 55](https://github.com/astral-sh/ruff/assets/58669/b38e6e74-db75-41ea-bcbe-6047ec1e8a2d)

![Screenshot 2023-06-14 at 10 55 08](https://github.com/astral-sh/ruff/assets/58669/19338295-40df-434c-8fce-4e75b69fa155)

**Comment by konstin** [comment]
This is really cool!
